diff --git a/core/iwasm/aot/aot_loader.c b/core/iwasm/aot/aot_loader.c
index 771ef87ff38c9a9c1fbfece13c427e736723e542..bac0b3c618947de55be8d3b386ba0a6df43a11ee 100644
--- a/core/iwasm/aot/aot_loader.c
+++ b/core/iwasm/aot/aot_loader.c
@@ -3911,15 +3911,21 @@ load_relocation_section(const uint8 *buf, const uint8 *buf_end,
         uint8 *mmap_addr = module->literal - sizeof(uint32);
         uint32 total_size =
             sizeof(uint32) + module->literal_size + module->code_size;
-        os_mprotect(mmap_addr, total_size, map_prot);
+        if (os_mprotect(mmap_addr, total_size, map_prot) != 0) {
+            set_error_buf(error_buf, error_buf_size, "protect AOT code failed");
+            goto fail;
+        }
     }
 
     map_prot = MMAP_PROT_READ;
 
 #if defined(BH_PLATFORM_WINDOWS)
     if (module->extra_plt_data) {
-        os_mprotect(module->extra_plt_data, module->extra_plt_data_size,
-                    map_prot);
+        if (os_mprotect(module->extra_plt_data, module->extra_plt_data_size,
+                        map_prot) != 0) {
+            set_error_buf(error_buf, error_buf_size, "protect AOT PLT failed");
+            goto fail;
+        }
     }
 #endif
 
@@ -3933,7 +3939,12 @@ load_relocation_section(const uint8 *buf, const uint8 *buf_end,
             /* ".rodata.strn.m" */
             || !strncmp(data_section->name, ".rodata.str",
                         strlen(".rodata.str"))) {
-            os_mprotect(data_section->data, data_section->size, map_prot);
+            if (os_mprotect(data_section->data, data_section->size, map_prot)
+                != 0) {
+                set_error_buf(error_buf, error_buf_size,
+                              "protect AOT data failed");
+                goto fail;
+            }
         }
     }
 
@@ -4373,8 +4384,13 @@ create_sections(AOTModule *module, const uint8 *buf, uint32 size,
                     destroy_aot_text = true;
 
                     if ((uint32)total_size > section->section_body_size) {
+#if (WASM_MEM_DUAL_BUS_MIRROR != 0)
+                        memset(mirrored_text + (uint32)section_size, 0,
+                               (uint32)total_size - section_size);
+#else
                         memset(aot_text + (uint32)section_size, 0,
                                (uint32)total_size - section_size);
+#endif
                         section->section_body_size = (uint32)total_size;
                     }
                 }
